scripts/main.js

```diff
@@ -0,0 +1,213 @@
+  image: null,
+  caption: null,
+  codeSnippet: null,
+  fetchedJson: null,
+};
+
+// ----------* Selecting nodes *-------------
+const imageInputElement = document.getElementById("image");
+
+// ----------* FUNCTION DECLARATIONS *--------------
+
+function renderGetCaptionButton() {
+  const getCaptionButton = getGetCaptionButton();
+  const main = document.querySelector('main');
+  main.appendChild(getCaptionButton);
+}
+
+// --
+
+function renderCaptionElement() {
+  const captionElement = getCaptionElement();
+  const div = document.createElement('div');
+  div.id = 'caption-container';
+  div.appendChild(captionElement);
+  const main = document.querySelector("main");
+  main.appendChild(div);
+}
+
+// ---
+
+function renderCopyButton() {
+  const copyButton = getCopyButton();
+  const div = document.querySelector("#caption-container");
+  div.appendChild(copyButton);
+}
+
+// ---
+
+function renderCopyImgCodeSnippetButton() {
+  const copyImgCodeSnippetButton = getCopyImgCodeSnippetButton();
+  const div = document.querySelector("#caption-container");
+  div.appendChild(copyImgCodeSnippetButton);
+}
+
+function renderImageSizeErrorModal() {
+  const main = document.querySelector('main');
+  const modal = getImageSizeErrorModal();
+  main.insertBefore(modal, main.firstChild);
+}
+
+// ---* helper functions *---
+function getGetCaptionButton() {
+  const button = document.createElement('button')
+  button.id = 'get-caption'
+  button.setAttribute('type', 'button');
+  button.innerText = 'Get Caption';
+
+  return button;
+}
+
+// --
+
+function renderImagePreview(selectedImageFile) {
+  if (!document.querySelector("img")) {
+    const imagePreview = document.createElement("img");
+    imagePreview.id = "image-preview";
+    imagePreview.setAttribute("src", "");
+    imagePreview.setAttribute("alt", "Image preview");
+    const div = document.querySelector("#image-preview-container");
+    div.appendChild(imagePreview);
+    const main = document.querySelector("main");
+    main.insertBefore(main.firstChild, div);
+  }
+
+  const imagePreview = document.querySelector("#image-preview");
+  const reader = new FileReader();
+
+  reader.addEventListener(
+    "load",
+    () => {
+      imagePreview.src = reader.result;
+    },
+    false
+  );
+
+  reader.readAsDataURL(selectedImageFile);
+}
+
+// ---
+
+function getCaptionElement() {
+  if (state.caption) {
+    const captionElement = document.createElement("p");
+    captionElement.innerText = `"${state.caption}"`;
+    return captionElement;
+  }
+}
+
+// ---
+
+function getCopyButton() {
+  const button = document.createElement("button");
+  button.setAttribute("type", "button");
+  button.innerText = "Copy caption text";
+
+  button.addEventListener("click", onCopyButtonClicked);
+
+  return button;
+}
+
+// ---
+
+function getImgCodeSnippet() {
+  const imagePreviewElement = document.querySelector("#image-preview");
+  const imageSource = imagePreviewElement.getAttribute("src");
```

Comment:
this seemed to produce a strange effect, the URL was very long and hard to read for me. I might recommend just defaulting to the end of the filename portion of the path, with the idea being that the user would update the path accordingly depending on where they actually add the image. this is definitely splitting hairs, and more the kind of thing to think of if you were developing this out further to be a real tool people would use